src/BrowserView.m

```diff
@@ -46,15 +46,29 @@ @interface BrowserView ()
 
 @property (weak, nonatomic) IBOutlet NSTabView *tabView;
 @property (weak, nonatomic) IBOutlet DisclosureView *tabBarDisclosureView;
-
+//queue for tab view items to select when current item is closed
+@property NSMutableArray<NSTabViewItem *> *tabViewOrder;
+//set to true to enable new tab selection behavior
+@property BOOL selectPreviousOnClose;
+@property BOOL selectNewItemFirst;
 @end
 
 @implementation BrowserView
 
 -(void)awakeFromNib
 {
 	[[self.tabView tabViewItemAtIndex:0] setLabel:NSLocalizedString(@"Articles", nil)];
-	
+
+	//TODO: make this a preference
+	self.selectPreviousOnClose = true;
+	self.selectNewItemFirst = true;
+
+	//do not initialize tabview order to restore default behavior
+	if (self.selectPreviousOnClose)
```

Comment:
Does it really change anything ? Whatever the option, we end up with an empty array…

Reply:
If the array is not initialized, every action on it will be ignored (which is a nice property of Objective-C). Thus the behavior with an uninitialized nil array is the old behavior, and when there is an initialized array, it is the new behavior.

Reply:
I understand, but if we want to translate the behavior to a dynamic preference, I think it's better to have a more readable code for future maintainers.